.github/workflows/main.yaml

```
@@ -0,0 +1,45 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches: [ master ]
+    branches_ignore: []
+
+jobs:
+  Linting:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Lint workflow
+      uses: snakemake/snakemake-github-action@v1.9.0
+      with:
+        directory: .
+        snakefile: workflow/Snakefile
+        args: "--lint"
+
+  Testing:
+    runs-on: ubuntu-latest
+    needs: Linting
+    steps:
+    - uses: actions/checkout@v1
+    - name: Checkout submodules
+      uses: textbook/git-checkout-submodule-action@2.0.0
+
+    - name: Test workflow (local test data)
+      uses: snakemake/snakemake-github-action@v1.9.0
+      with:
+        directory: .test
+        snakefile: workflow/Snakefile
+        args: "--use-conda --cache get_genome get_annotation genome_faidx --show-failed-logs -j 10 --conda-cleanup-pkgs cache --conda-frontend mamba"
```

Here, you only need to specify the `--cache` flag, not the rule names, as the rule definitions already have `cache: true` set, see e.g. the [example for the get_genome rule](https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/reference/ensembl-sequence.html#example) or [the respective rule version in this PR](https://github.com/snakemake-workflows/chipseq/pull/4/files#diff-9d2f60323f47caa8ed15dd7ff62d1ccdR11). The [snakemake docs about in between workflow caching](https://snakemake.readthedocs.io/en/latest/executing/caching.html#between-workflow-caching) (look a bit further down) then says, that you only have to set the flag without the rule names given -- the command line help is a bit confusing regarding this.

```suggestion
        args: "--use-conda --cache --show-failed-logs -j 10 --conda-cleanup-pkgs cache --conda-frontend mamba"
```